src/ServerProviders/Nginx/Plugins.ServerProviders.Nginx/ServerProviderNginx.cs

```diff
@@ -109,31 +111,42 @@ private string GetShellCommandOutput(string command)
 
             process.ErrorDataReceived += (obj, a) =>
             {
-                if (!string.IsNullOrWhiteSpace(a.Data))
+                if (!String.IsNullOrWhiteSpace(a.Data))
                 {
                     _log?.Error($"Error: {a.Data}");
+                    errorOutput += a.Data;
                 }
             };
 
             try
             {
-                var timeoutMS = 1000 * 5;
-
                 process.Start();
 
                 process.BeginOutputReadLine();
                 process.BeginErrorReadLine();
 
-                process.WaitForExit(timeoutMS);
+                // It is noted in the .NET API documentation that when using asynchronous output streams for Process,
+                // it is best to call WaitForExit(), as specifying a timeout by using the WaitForExit(Int32) overload
+                // does not ensure the output buffer has been flushed. 
+                // See https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.errordatareceived?view=net-7.0#remarks
+                process.WaitForExit();
```

Comment:
That's cool, but we can't risk waiting indefinitely.

Reply:
I will restore the previous version